tests/integration/optlyplugins/proxy_dispatcher.go

```diff
@@ -0,0 +1,32 @@
+package optlyplugins
+
+import (
+	"time"
+
+	"github.com/optimizely/go-sdk/pkg/event"
+)
+
+// EventReceiver returns dispatched events
+type EventReceiver interface {
+	GetEvents() []event.Batch
+}
+
+// ProxyEventDispatcher represents a valid HTTP implementation of the Dispatcher interface
+type ProxyEventDispatcher struct {
+	events []event.Batch
+}
+
+// DispatchEvent dispatches event with callback
+func (d *ProxyEventDispatcher) DispatchEvent(event event.LogEvent) (bool, error) {
+	d.events = append(d.events, event.Event)
+	return true, nil
+}
+
+// GetEvents returns dispatched events
+func (d *ProxyEventDispatcher) GetEvents() []event.Batch {
+	time.Sleep(600 * time.Millisecond)
```

Comment:
Why is this sleep necessary? Seems like a source of flakiness. Let's find a way around it